prime-router/src/main/kotlin/Report.kt

```diff
@@ -61,9 +61,20 @@ enum class Options {
     ValidatePayload,
     CheckConnections,
     SkipSend,
-    SendImmediately;
+    SendImmediately,
+    @OptionDeprecated
+    SkipInvalidItems;
+
+    /**
+     * Checks to see if the enum constant has an @OptionDeprecated annotation.
+     * If the annotation is present, the constant is no longer in use.
+     */
+    fun isDeprecated(): Boolean = this.declaringClass.getField(this.name)
```

Comment:
One more.  To be more like Kotlin, change this to be a variable using get().  I am thinking you have not seen that yet.  The call will look like `options.isDeprecated` instead of using `()`.  Just more kotlin like.  See https://kotlinlang.org/docs/properties.html#getters-and-setters